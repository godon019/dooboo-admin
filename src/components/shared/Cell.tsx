import styled, { CSSProp } from 'styled-components';

import { LabelSize } from '../../utils/Styles';
import React from 'react';

interface CellProps {
  label: string;
  htmlFor?: string;
  children: React.ReactNode;
  labelCss?: CSSProp;
  errors: any;
}
const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 24px;
`;
const LabelAndContent = styled.div`
  display: flex;
  align-items: center;
`;

const LabelSpan = styled.label<{ labelCss: CSSProp }>`
  ${LabelSize}
  font-size: 14px;
  text-align: left;
  ${(props) => props.labelCss};
  color: ${(props) => props.theme.fontColor};
`;

const Error = styled.div`
  align-self: center;
  margin-top: 0.1em;
  /* margin-left: 12em; */
  color: red;
  font-size: 0.8em;
`;

const Cell: React.FC<CellProps> = ({
  label,
  children,
  htmlFor,
  labelCss,
  errors,
  ...rest
}) => {
  return (
    <Container>
      <LabelAndContent {...rest}>
        <LabelSpan htmlFor={htmlFor} labelCss={labelCss}>
          {label}
        </LabelSpan>
        {children}
      </LabelAndContent>
      {/* <Error>{'what'}</Error> */}
      {errors && errors[htmlFor] && <Error>{errors[htmlFor]}</Error>}
    </Container>
  );
};

export default Cell;
