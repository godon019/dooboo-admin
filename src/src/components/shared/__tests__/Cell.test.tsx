import React from 'react';
import renderer from 'react-test-renderer';

import Cell from '../Cell';
import { render } from '@testing-library/react';

let props: any;

props = {
  label: 'name',
};

let rendered;
let utils;

beforeAll(() => {
  rendered = renderer.create(<Cell {...props}>nick</Cell>).toJSON();
  utils = render(<Cell {...props}>nick</Cell>);
});

describe('[Cell] render', () => {
  it('renders without crashing', () => {
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it('should should children element', () => {
    utils.getByText('nick');
  });
});
