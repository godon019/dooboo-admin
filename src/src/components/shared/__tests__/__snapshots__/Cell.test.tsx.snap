// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Cell] render renders without crashing 1`] = `
<div
  className="sc-bdVaJa jGfYvm"
>
  <label
    className="sc-bwzfXH fPyPqn"
  >
    name
  </label>
  nick
</div>
`;
